text/0000-deprecate-string-based-actions.md

```diff
@@ -0,0 +1,89 @@
+- Start Date: 2020-05-23
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# Deprecate String Based Actions
+
+## Motivation
+
+> In Ember 1.13 we introduced [closure actions](https://github.com/emberjs/rfcs/blob/00ac2685c86f27d41547012903f485a4ef338d27/active/0000-improved-actions.md) have been recommended over string based actions. This officially deprecates string based actions in favor of closure actions,
+which provide many benefits such as
+
+1. simpler passing through components
+2. less need for an actions hash
+3. ability to return a value
+4. typing in TypeScript
+5. compatible with Glimmer Components
```

Comment:
I'm not sure this is actually true, have you tested? I'm **pretty sure** you can use `@action whatever() {}` in a `@glimmer/component` and then use something like `<button {{action 'whatever'}}></button>` just fine.

Reply:
I wasn't aware of this. I thought a quoted action required the actions hash. TIL.

Reply:
Well, it _does_ require an actions hash. The `@action` decorator (either in decorator form or computed style syntax) **creates** an actions hash for compatibility purposes.  


https://github.com/emberjs/rfcs/blob/master/text/0408-decorators.md#actions has more details on how that works, a small snippet:

> This would be implemented by creating the class's actions object and assigning the method to it, setting up inheritance and such in the process. The decorator leaves the method definition on the class, where it can be called like a normal method would, including super functionality. This maintains compatibility with the classic object model, while making the actions namespace an implementation detail rather than something users need to know about.

Reply:
This means that there is no need for an actions hash in the previous discussion, right?

Reply:
I read that (in the list above right?) as referring to you _the user_ specifying an actions hash.

Reply:
I'm not sure if it is possible for a deprecation warning to distinguish between the user providing an actions hash and Ember itself providing it?

Reply:
Ya, should be possible. Happy to spit ball specific implementation details to make that work, but internally when Ember merges mixins (the POJO passed to `.extend()` is technically treated as a mixin) we can handle explicit `actions` property there differently than the `action` helper imperatively creating it.